src/Symfony/Bundle/FrameworkBundle/Controller/AbstractController.php

```diff
@@ -290,6 +290,42 @@ protected function stream(string $view, array $parameters = [], StreamedResponse
         return $response;
     }
 
+    /**
+     * Handles a form:
+     *
+     * * if the form is not submitted, the Twig template passed in $view is rendered and a 200 HTTP status code is set
+     * * if the form is submitted but invalid, the Twig template passed in $view is rendered and 422 HTTP status code is set
+     * * if the form is submitted and valid, the entity is saved (only if it is managed by Doctrine ORM), a 306 HTTP status code is set and the Location HTTP header is set to the value of $redirectUrl
+     */
+    public function handleForm(Request $request, FormInterface $form, string $view, ?string $redirectUrl = null, ?callable $onSuccess = null, array $viewParameters = [], ?Response $response = null): Response
```

Comment:
Having to generate the redirectUrl before handling the form forbids relying on anything that depends on the processing (so no slug in the URL parameter as the form is not bound yet, and no DB-generated id as the data is not persisted yet)